cluster/v1alpha1/types.go

```diff
@@ -301,32 +301,79 @@ const (
+	// If both Name and PrioritizerScoreCoordinate.Builtin are defined, will use the value
+	// in PrioritizerScoreCoordinate.Builtin.
 	// Name is the name of a prioritizer. Below are the valid names:
 	// 1) Balance: balance the decisions among the clusters.
 	// 2) Steady: ensure the existing decision is stabilized.
 	// 3) ResourceAllocatableCPU & ResourceAllocatableMemory: sort clusters based on the allocatable.
-	// +kubebuilder:validation:Required
-	// +required
-	Name string `json:"name"`
+	// +optional
+	Name string `json:"name,omitempty"`
 
-	// Weight defines the weight of prioritizer. The value must be ranged in [0,10].
+	// PrioritizerScoreCoordinate represents the configuration of the prioritizer and score source.
+	// +optional
+	PrioritizerScoreCoordinate *PrioritizerScoreCoordinate `json:"scoreCoordinate,omitempty"`
+
+	// Weight defines the weight of the prioritizer score. The value must be ranged in [-10,10].
 	// Each prioritizer will calculate an integer score of a cluster in the range of [-100, 100].
 	// The final score of a cluster will be sum(weight * prioritizer_score).
 	// A higher weight indicates that the prioritizer weights more in the cluster selection,
-	// while 0 weight indicate thats the prioritizer is disabled.
-	// +kubebuilder:validation:Minimum:=0
+	// while 0 weight indicates that the prioritizer is disabled. A negative weight indicates
+	// wants to select the last ones.
+	// +kubebuilder:validation:Minimum:=-10
 	// +kubebuilder:validation:Maximum:=10
 	// +kubebuilder:default:=1
 	// +optional
 	Weight int32 `json:"weight,omitempty"`
 }
 
+// PrioritizerScoreCoordinate represents the configuration of the prioritizer and score source
+type PrioritizerScoreCoordinate struct {
+	// Type defines the type of the prioritizer.
+	// Type is either "Builtin", "AddOn" or "", where "" is "Builtin" by default.
+	// When the type is "Builtin", need to specify a Builtin prioritizer name in Builtin.
+	// When the type is "AddOn", need to configure the score source in AddOn.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=Builtin;AddOn
+	// +kubebuilder:default:=Builtin
+	// +required
+	Type string `json:"type,omitempty"`
+
+	// Builtin defines the name of a builtin prioritizer. Below are the valid builtin prioritizer names.
+	// 1) Balance: balance the decisions among the clusters.
+	// 2) Steady: ensure the existing decision is stabilized.
+	// 3) ResourceAllocatableCPU & ResourceAllocatableMemory: sort clusters based on the allocatable.
+	// +optional
+	Builtin string `json:"builtin,omitempty"`
+
+	// When type is "AddOn", AddOn defines the resource name and score name.
+	// +optional
+	AddOn *PrioritizerAddOnScore `json:"addOn,omitempty"`
+}
+
+// PrioritizerAddOnScore represents the configuration of the addon score source.
+type PrioritizerAddOnScore struct {
+	// ResourceName defines the resource name of the AddOnPlacementScore.
```

Comment:
resourceNames are usually actually `resource` and `group`.  For clarity in a single key, I'm willing to try a fully qualified resource name (that is `resource.group`) in an alpha API if you and the other reviewers here would like to.